sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1847,3 +1848,25 @@ object OptimizeLimitZero extends Rule[LogicalPlan] {
       empty(ll)
   }
 }
+
+/**
+ * Generates filters for exploded expression, such that rows that would have been removed
+ * by this [[Generate]] can be removed earlier - before joins and in data sources.
+ */
+object InferFiltersFromGenerate extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
+    case g @ Generate(e: ExplodeBase, _, false, _, _, child)
```

Comment:
How about the other gerators, e.g., `Inline`?

Reply:
I'm not familiar with other generators and focused only on the ones that I regularly use. 
But yes, I'm sure it can be expanded to include other Generators as well, but there can be some exceptions.

Reply:
Yea, its better to conver the otehr cases where pssible, too:
https://github.com/apache/spark/blob/0812d6c17cc4876bb87a9d1fec35ec8c7b2365f0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala#L432-L434

Reply:
Added `Inline` and also checked other `Generate`-s, but it seems that this rule is not applicable to others.

Reply:
Looks okay. Could you update the PR description accordingly?

Reply:
Sure